addons/udes_stock/models/stock_location.py

```diff
@@ -428,20 +428,53 @@ def _get_stock_drift(self, adjustments_request):
         Product = self.env['product.product']
         Package = self.env['stock.quant.package']
         Lot = self.env['stock.production.lot']
-
+        wh = self.env.user.get_user_warehouse()
         stock_drift = {}
         new_packages = {}
 
+        # Group quantities by product and reserved packages
+        no_package_vals = defaultdict(int)
+        adj_reqs = []
+        for adj in adjustments_request:
+            pn = adj['package_name']
+            if NO_PACKAGE_TOKEN in pn or pn in wh.reserved_package_name:
+                no_package_vals[adj['product_id']] += adj['quantity']
+            else:
+                adj_reqs.append(adj)
+
+        # Replace requests with values passed through by reserved filter
+        adjustments_request = adj_reqs
+
+        # Readd them to adjustments_request
+        new_adjs = []
+        for product, quantity in no_package_vals.items():
+            new_adjs.append({
+            'product_id': product,
+            'package_name': 'NO_PACKAGE',
+            'quantity': quantity,
+            })
+        adjustments_request.extend(new_adjs)
+
+        # Look through all quants not in packages within location and index
+        # the quantities by product, as we have collected quantities by product
+        # for the adjustments without package this allows us to directly compare
+        # and skip unchanged loose product quantities
+        products_and_quantities = {}
+        for quant in self.mapped('quant_ids'):
+            if not quant.package_id:
+                products_and_quantities[quant.product_id] = quant.quantity
+
+        # Go through (potentially modified) adjustments_request
         for adj in adjustments_request:
             product = Product.get_product(int(adj['product_id']))
-
             # determine the package
-
             package_name = adj['package_name']
             package = None
-
             if NO_PACKAGE_TOKEN in package_name:
-                pass
+                if products_and_quantities.get(product) == adj['quantity']:
+                    continue
+                else:
```

Comment:
no need for the `else: pass`

Reply:
so i should just combine the booleans right

Reply:
oops just read it back, that would break with the elif and else